dev/TabView/TabViewItem.cpp

```diff
@@ -110,6 +122,16 @@ void TabViewItem::OnLoaded(const winrt::IInspectable& sender, const winrt::Route
     }
 }
 
+void TabViewItem::OnCommonVisualStateChanged(const winrt::IInspectable& sender, const winrt::VisualStateChangedEventArgs& args)
+{
+    // our base class ListViewItem manages Disabled visual state in separate group, but we placed it into common states, and need to adjust for this
+    const auto newStateName = args.NewState().Name();
+    if (!IsEnabled() && newStateName != L"SelectedDisabled" && newStateName != L"Disabled")
+    {
+        winrt::VisualStateManager::GoToState(*this, IsSelected() ? L"SelectedDisabled" : L"Disabled", false);
```

Comment:
This is a pretty creative solution. I've been wracking my brain (and asking around) if there are any things we expect to go wrong from this and I can't come up with anything.  @chingucoding did you see things? any concerns?

I like the work around.

Reply:
I'm not sure if the if case condition is correctly or if I am missing something here. What exactly is the idea? If the control is getting any common state, we want to force it into the corresponding disabled state?

Reply:
Exactly

Reply:
I see, thanks. While I would prefer to not interfere with the state setting, I guess this is fine here, the only other solutions I can think of are even worse. Of course once TabView moves to ItemsRepeater this code is probably not needed anymore but that's a different discussion.
TL;DR: Looks good to me.